File: src/main/scala/com/kapilkoju/autopos/trade/purchase/domain/PurchaseLine.scala
package com.kapilkoju.autopos.trade.purchase.domain

import java.lang.Long
import java.math.BigDecimal
import javax.persistence._
import javax.validation.constraints.{Min, NotNull}

import com.fasterxml.jackson.annotation.JsonBackReference
import com.kapilkoju.autopos.trade.domain.LineItem

import scala.beans.BeanProperty

@Entity
@Table(name = "purchase_line")
class PurchaseLine extends LineItem {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @BeanProperty
  var id: Long = _

  @NotNull
  @Min(value = 0)
  @Column(name = "rate", precision = 10, scale = 2, nullable = false)
  @BeanProperty
  var rate: BigDecimal = _

  @NotNull
  @ManyToOne(optional = false)
  @JoinColumn(name = "purchase_id")
  @JsonBackReference
  @BeanProperty
  var purchase: Purchase = _
}

File: kotlang/src/main/java/com/kapilkoju/autopos/trade/purchase/domain/PurchaseLine.java
package com.kapilkoju.autopos.trade.purchase.domain;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.kapilkoju.autopos.trade.domain.LineItem;

import javax.persistence.*;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;


@Entity
@Table(name = "purchase_line")
public class PurchaseLine extends LineItem {

    @NotNull
    @Min(value = 0)
    @Column(name = "rate", precision = 10, scale = 2, nullable = false)
    private BigDecimal rate;

    @NotNull
    @ManyToOne(optional = false)
    @JoinColumn(name = "purchase_id")
    @JsonBackReference
    private Purchase purchase;

    @Override
    public BigDecimal getRate() {
        return rate;
    }

    public void setRate(BigDecimal rate) {
        this.rate = rate;
    }

    public Purchase getPurchase() {
        return purchase;
    }

    public void setPurchase(Purchase purchase) {
        this.purchase = purchase;
    }
}
